From 2ad5f1a8c789ee93df38a832254dae1a02d68759 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jollamobile.com>
Date: Tue, 19 Sep 2017 17:56:14 +0300
Subject: [PATCH 65/65] [PackageKit] Accept also overridden architecture for
 local-install packages. Contributes to JB#39341

Aarch64 overridden as armv7hl etc.
---
 backends/zypp/pk-backend-zypp.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index cd51da92c..c1d95a89d 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -3008,6 +3008,7 @@ backend_install_files_thread (PkBackendJob *job, GVariant *params, gpointer user
 
 		// check if the file has an acceptable architecture
 		if (rpmHeader->tag_arch() != ZConfig::defaultSystemArchitecture () &&
+				rpmHeader->tag_arch() != ZConfig::instance().systemArchitecture () &&
 				rpmHeader->tag_arch() != Arch_noarch &&
 				! system_and_package_are_x86 (rpmHeader->tag_arch ())) {
 			zypp_backend_finished_error (
-- 
2.13.5

